README.md

```diff
@@ -809,20 +620,367 @@ When you have time, we invite you to take a look at:
+
+* insert the orders within a Database
+* notify the delivery company using a specific API (This one will be provided during the session)
+
+The application is designed around Event Driven Architecture and uses Kafka to handle those messages:
+* The coffee orders to be processed are in a kafka topic named `orders`
+* The notifications to the delivery company are in a kafka topic named `deliveries`
+
+The application that create orders and notify delivery teams are external to our ecosystem.
+
+![](resources/CQ-workshop1.png)
+
+Let's navigate to the part 5 folder.
+
+### Dev Services
+Quarkus supports the automatic provisioning of unconfigured services in development and test mode. For more information, you can check the [official Dev Services overview](https://quarkus.io/guides/dev-services).
+As we are using kafka and PostgreSQL, without any configuration, Quarkus dev services will start automatically Strimzi and PostgreSQL. 
+Start the app in dev mode. Notice that those 2 servers have been started.
+
+### Generate coffee orders
+In this part we use a Camel Route to simulate the external app that generates some coffee orders. Those orders are pushed to the Kafka topic named `orders`.
+
+To achieve this, we will get random coffees from an [external API](https://random-data-api.com/api/coffee/random_coffee) that generates Random coffee. We use a timer to call the API every 1s.
+We use 2 components : [timer](https://camel.apache.org/components/latest/timer-component.html) and [http](https://camel.apache.org/components/latest/http-component.html), as described in [MyRoutes class](part-5-kafka/src/main/java/org/acme/MyRoutes.java):
+```java
+public class MyRoutes extends RouteBuilder {
+
+    @Override
+    public void configure() throws Exception {
+        from("timer:create-random-coffee-orders")
+                .to("http:{{random-coffee-api}}");
+    }
+}
+```
+`random-coffee-api` value is described in the [application.properties configuration file](part-5-kafka/src/main/resources/application.properties). It contains the URI to the random coffee API.
+The response from the API is in json format. We use the [Jackson is a Data Format](https://camel.apache.org/components/latest/dataformats/json-jackson-dataformat.html) to convert from Json to [Coffee Java class](part-5-kafka/src/main/java/org/acme/Coffee.java).
+
+```
+    .unmarshal().json(Coffee.class)
+```
+
+Then we use [bean component](https://camel.apache.org/components/latest/bean-component.html) to transform the body to a [Coffee Order](part-5-kafka/src/main/java/org/acme/CoffeeOrder.java). Note that we do specify the method everytime the bean class contains many methods.
+```
```

Comment:
"Note that we do specify the method everytime the bean class contains many methods."
=>
"Note that we explicitly specify the method name `generateOrder` to select amongth the `myBean` methods."